tutorials/llm/llama-3/nemo2-sft-peft/nemo2-peft.ipynb

```diff
@@ -0,0 +1,540 @@
+    "\n",
+    "## Data\n",
+    "This notebook uses the SQUAD dataset. For more details about the data refer to [SQuAD: 100,000+ Questions for Machine Comprehension of Text](https://arxiv.org/abs/1606.05250)\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "# Step 1: Import HuggingFace checkpoint\n",
+    "We use the `llm.import_ckpt` API to download the specified model using the \"hf://<huggingface_model_id>\" URL format. It will then convert the model into NeMo 2.0 format. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "import nemo_run as run\n",
+    "from nemo import lightning as nl\n",
+    "from nemo.collections import llm\n",
+    "from megatron.core.optimizer import OptimizerConfig\n",
+    "from nemo.collections.llm.peft.lora import LoRA\n",
+    "import torch\n",
+    "import pytorch_lightning as pl\n",
+    "from pathlib import Path\n",
+    "from nemo.collections.llm.recipes.precision.mixed_precision import bf16_mixed\n",
+    "\n",
+    "\n",
+    "# llm.import_ckpt is the nemo2 API for converting Hugging Face checkpoint to NeMo format\n",
+    "# example usage:\n",
+    "# llm.import_ckpt(model=llm.llama3_8b.model(), source=\"hf://meta-llama/Meta-Llama-3-8B\")\n",
+    "#\n",
+    "# We use run.Partial to configure this function\n",
+    "def configure_checkpoint_conversion():\n",
+    "    return run.Partial(\n",
+    "        llm.import_ckpt,\n",
+    "        model=llm.llama3_8b.model(),\n",
+    "        source=\"hf://meta-llama/Meta-Llama-3-8B\",\n",
+    "        overwrite=False,\n",
+    "    )\n",
+    "\n",
+    "# configure your function\n",
+    "import_ckpt = configure_checkpoint_conversion()\n",
+    "# define your executor\n",
+    "local_executor = run.LocalExecutor()\n",
+    "\n",
+    "# run your experiment\n",
+    "run.run(import_ckpt, executor=local_executor)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 2: Prepare data\n",
+    "\n",
+    "We will be using SQUAD for this notebook. NeMo 2.0 already provides a `SquadDataModule`. Example usage:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "def squad() -> run.Config[pl.LightningDataModule]:\n",
+    "    return run.Config(llm.SquadDataModule, seq_length=2048, micro_batch_size=1, global_batch_size=8, num_workers=0)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For how to use your own data to create your custom `DataModule` in order to perform PEFT, refer to [NeMo 2.0 SFT notebook](./nemo2-sft.ipynb)."
```

Comment:
revise

To learn how to use your own data to create a custom DataModule for performing PEFT, refer to [NeMo 2.0 SFT notebook](https://github.com/NVIDIA/NeMo/blob/a5644b10e403580f750889e9bed16c121b10552a/tutorials/llm/llama-3/nemo2-sft-peft/nemo2-sft.ipynb).

Reply:
revised. thanks